code/frameworks/react-webpack5/src/preset.ts

```diff
@@ -9,46 +9,14 @@ export const addons: PresetProperty<'addons'> = [
   getAbsolutePath('@storybook/preset-react-webpack'),
 ];
 
-const defaultFrameworkOptions: FrameworkOptions = {
-  legacyRootApi: true,
-};
-
-export const frameworkOptions = async (
-  _: never,
-  options: Options
-): Promise<StorybookConfig['framework']> => {
-  const config = await options.presets.apply<StorybookConfig['framework']>('framework');
-
-  if (typeof config === 'string') {
-    return {
-      name: config,
-      options: defaultFrameworkOptions,
-    };
-  }
-  if (typeof config === 'undefined') {
-    return {
-      name: getAbsolutePath('@storybook/react-webpack5'),
-      options: defaultFrameworkOptions,
-    };
-  }
-
-  return {
-    name: config.name,
-    options: {
-      ...defaultFrameworkOptions,
-      ...config.options,
-    },
-  };
-};
-
 export const core: PresetProperty<'core'> = async (config, options) => {
-  const framework = await options.presets.apply('framework');
+  const presetFramework = await options.presets.apply('framework');
 
   return {
     ...config,
     builder: {
       name: getAbsolutePath('@storybook/builder-webpack5'),
-      options: typeof framework === 'string' ? {} : framework.options.builder || {},
+      options: typeof presetFramework === 'string' ? {} : presetFramework.options.builder || {},
```

Comment:
If you completely removed the `framework` export above, then you don't need to rename this. It's good not to rename this because there are many other places that use this exact same code, and if we ever refactor them we probably would miss this. Or you could rename every place that has this, so they are "find-replaceable"